out/automation/util/log-listener.js

```diff
@@ -1,33 +1,48 @@
-/**  Listen to port 1 and logs all values received.
- *    Strings written out which begin and end with bracktes are assumed to be json objects.
- *    Json printing also prints to a single file with all json objects and to individual log files for each host.
+/**
+ * TODO(juxtaposedwords): find a way to make the need for 'await'  not necesasry when calling
+ * TODO(juxtaposedwords): Add logging levels
+ * TODO(juxtaposedwords): Add log file line truaction.
+ * Json-logger creates a listener which listens to port 2 and writes everything received to a file asssociated with the logging services' name.
+ *  Usage example:
+ *      export async function main(ns) {
+ *          await log(ns, "i'be been waiting.");
+ *          await log(ns, "for the longest time.");
+ *      }
+ *  async function log(ns, message) { await jsonLog(ns, "MY_COMPLETLEY_REAL_PROGRAM_NAME", message)} 
  * 
- * @param {import("../../..").NS } ns */
+ *  @param {import("../../..").NS } ns */
 export async function main(ns) {
-    ns.disableLog('ALL')
-    const h = ns.getPortHandle(1);
+    const data = ns.flags([
+		["port", 2], // Which port the listener should listen to.
+        ["log_destination", "home"], // Which port the listener should listen to.
+	]);
+    const port = ns.getPortHandle(data["port"]);
```

Comment:
flags.port, not data["port"].  all keys in JS objects are accessible as properties.

Reply:
done